chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -1,53 +1,93 @@
 package org.reactome.release.chebiupdate;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.gk.model.GKInstance;
 import org.gk.model.InstanceDisplayNameGenerator;
-import org.gk.model.PersistenceAdaptor;
 import org.gk.model.ReactomeJavaConstants;
 import org.gk.persistence.MySQLAdaptor;
 import org.gk.persistence.MySQLAdaptor.AttributeQueryRequest;
 import org.gk.schema.InvalidAttributeException;
 import org.gk.schema.InvalidAttributeValueException;
-import org.gk.schema.SchemaClass;
-import org.gk.util.GKApplicationUtilities;
+import org.reactome.release.common.database.InstanceEditUtils;
 
 import uk.ac.ebi.chebi.webapps.chebiWS.client.ChebiWebServiceClient;
 import uk.ac.ebi.chebi.webapps.chebiWS.model.ChebiWebServiceFault_Exception;
 import uk.ac.ebi.chebi.webapps.chebiWS.model.DataItem;
 import uk.ac.ebi.chebi.webapps.chebiWS.model.Entity;
 
-
-public class ChebiUpdater {
+public class ChebiUpdater
+{
 	private static final Logger logger = LogManager.getLogger();
+	private static final Logger refMolNameChangeLog = LogManager.getLogger("molNameChangeLog");
+	private static final Logger refMolIdentChangeLog = LogManager.getLogger("molIdentChangeLog");
+	private static final Logger refEntChangeLog = LogManager.getLogger("refEntChangeLog");
+	private static final Logger duplicatesLog = LogManager.getLogger("duplicatesLog");
+	private static final Logger failedChebiLookupsLog = LogManager.getLogger("failedChebiLookupsLog");
 	private boolean testMode = true;
 	private MySQLAdaptor adaptor;
 	private ChebiWebServiceClient chebiClient = new ChebiWebServiceClient();
-	private StringBuilder identifierSB = new StringBuilder();
 	private StringBuilder formulaUpdateSB = new StringBuilder();
 	private StringBuilder formulaFillSB = new StringBuilder();
-	private StringBuilder nameSB = new StringBuilder();
 	private StringBuilder duplicatesSB = new StringBuilder();
-
+	private Map<GKInstance, List<String>> referenceEntityChanges = new HashMap<GKInstance, List<String>>();
 	private long personID;
+	private boolean useCache;
+	// A Comparator object that will compare GKInstances, assuming that they are of the "Person" type, with a surname and firstname.
+	private Comparator<GKInstance> personComparator = new Comparator<GKInstance>()
```

Comment:
Maybe move the initialization of the personComparator to its own method so the constructor is immediately after the field declarations